### ISSUE TITLE
improve behaveour of NVDA in QT applications such as MuseScore, Virtualbox, VLC, Caliber and others

### ISSUE THREAD

**Issue by gregjozk** [nvaccess/nvda#14756]
### Is your feature request related to a problem? Please describe.

It seems, that NVDA has many problems with Qt applications such as MuseScore, VLC, Caliber, Virtualbox and others.
e.g. nvda+up (say current line and other shortcuts don't work, object navigation is useless, NVDA does not report, what is selected or checked if it is etc

### Describe the solution you'd like

I wish only, that I would use these applications as smoothly as possible. because I do not know much about coding, I cannot provide possible solutions. but I think, that that colaboration between Qt team and NVDA team would help due to complexity of problems and perhaps work will have to be done on both sides.

### Additional context

Here are several issues about Qt, which are opened for one of Qt problems, but with this issue I want to possibly open a discussion if and how Qt problems could be solved. e.g., when used MuseScore, even MuseScore team worked on accessibility, NVDA and MuseScore are not total compatible. Try to write some scores, see accessibility videos, where some problems like "not selected", silence and others are presented.

thanks.

**Comment by jcsteh** [comment]
This issue is far too broad to be actionable. Even if there is to be collaboration with the Qt team, there need to be clearly documented scenarios and steps to reproduce. So, rather than saying that read line doesn't work, you would need to provide exact steps to reproduce in a specific application, along with what you expected and what happened instead. These should also be filed as separate issues. Read line, check boxes not working and object navigation are very separate issues which will need very separate fixes, regardless of where those fixes need to land.

Thanks for your time and willingness to improve accessibility across the board.

**Comment by gregjozk** [comment]
Hello,

I understand and respect that developers prefer issues (one
problem/one issue) with STR and logs, but whenever I've used Qt apps,
there were same problems and there are more than one. Maybe some could
be app spacific, but others such as "read line" are general.

read line:

- open one Qt app such as MuseScore
- press ins+up

you always hear "blank". It happens in all tested Qt apps.

object navigation also does not help. whenever I wanted to find some
commands, it is not possible. In VLC it is somewhat usable though, but
in Musescore e.g., it is unusable, because NVDA returns me "system" or
goes outside of Musescore window, so parent/child relationship is very
difficult to research.

Can I suggest instead of closing, that this issue is converted to
discussion due to popularity of Qt framework. I see that more and more
developers use it for mostly cross-platform apps.

thanks.

2023-03-29 4:14 GMT+02.00, James Teh ***@***.***>:
